content/en/docs/concepts/overview/working-with-objects/storage-version.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Storage Versions
+content_type: concept
+weight: 20
+---
+
+<!-- overview -->
+Storage version is the current version that the API object is stored at in the
+cluster. While it does not affect what the API Server returns it is an important
+concept to understand since it is what links the API representation of the
+object to the actual encoding in the storage backend. This can matter for when
+the underlying binary encodings of the object matter, such as for encryption at
+rest, or API deprecation.
+
+The same API object may have multiple storage versions that the API Server can
+then convert to an object schema. This means that the API Server is aware of the
+binary encodings of the objects and is able to convert between all the stored
+versions to the API Representation of the object dynamically.
+
+The version of an object is separate from the storage version entirely. For
+example, a `v1alpha1` and `v1beta1` API Object for the same GroupResource will
+be encoded the same in storage as long as the storage version has not been
+updated between the two objects.
+
+<!-- body -->
+
+## Storage Version to Resource Mapping
+
+Every resource will have 1 active storage version at any point in time, meaning
+that any write to an object will store the object at that storage version. Reads
+can occur from any storage version however, and will not update the stored
+version of the object. The only way a storage version will get updated is by
+modifying the object somehow. Otherwise the stored version will stay the same as
+long as the object exists in the cluster. 
+
+## Encryption at Rest
+
+There are tools to [encrypt the at rest
+storage](/docs/tasks/administer-cluster/kms-provider/) of a cluster, especially
+for cluster secrets. This adds an additional layer of protection for data
+exfiltration since the actual stored data in the cluster is encrypted. This
+means that the API Server is actually decrypting the stored data when it decodes
+the data from storage. The APIServer must have the key for that storage version
+in order to decode the object properly.
```

Comment:
The key is not actually stored per storage version, but you may want to migrate upon a key-encryption-key rotation. @enj or @aramase please correct me if this is still inaccurate.

Reply:
Yeah, not as familiar with encryption as with the multiple API versions, correct me if I have any misunderstandings here. I was under the impression that the key couldn't be deprecated until all objects of the resource had been updated since otherwise the object may be stored encrypted with the old key.

Reply:
If you remove access to the key while there are some objects still encrypted with it, you lose access to those objects, that's indeed correct.

Reply:
Sounds good, resolving!